packages/vm/test/t8n/t8ntool.ts

```diff
@@ -164,43 +164,78 @@ export class TransitionTool {
 
     let index = 0
 
-    this.vm.events.on('afterTx', (event) => {
-      this.afterTx(event, index, builder)
-    })
+    let trace: any[] = []
+    // Tracing
+    if (args.trace === true) {
+      this.vm.evm.events?.on('step', (e) => {
+        let hexStack = []
+        hexStack = e.stack.map((item: bigint) => {
+          return '0x' + item.toString(16)
+        })
+        const opTrace = {
+          pc: e.pc,
+          op: e.opcode.name,
+          gas: bigIntToHex(e.gasLeft),
+          gasCost: bigIntToHex(BigInt(e.opcode.fee)),
+          stack: hexStack,
+          depth: e.depth,
+          opName: e.opcode.name,
+        }
+        trace.push(JSON.stringify(opTrace))
+      })
+      this.vm.events.on('afterTx', async (event) => {
+        const summary = {
+          stateRoot: bytesToHex(await this.vm.stateManager.getStateRoot()),
+          output: bytesToHex(event.execResult.returnValue),
+          gasUsed: bigIntToHex(event.totalGasSpent),
+          pass: event.execResult.exceptionError === undefined,
+          fork: this.vm.common.hardfork(),
+        }
+        trace.push(JSON.stringify(summary))
+        writeFileSync(
+          `trace-${index}-${bytesToHex(event.transaction.hash())}.json`,
```

Comment:
I think this should be written to the output dir `args.output.basedir` (maybe even in a dedicated trace folder, take a look what Geth does in this case if they have a specific trace folder or just dump all in the output dir). Also note that in the current setup it will depend on from which path the t8ntool is being called (will write to current path, not something relative to it).